### PR TITLE
Item update with unittest

### PR DESCRIPTION
**Pull request by ankitdas13**
command for runnning api from sdk folder : `bundle exec ruby test.rb`
command for runnning testing : `bundle exec rake test`

for create : https://github.com/razorpay/razorpay-node/blob/master/documents/items.md#create-item
Link : https://razorpay.com/docs/api/items/#create-an-item

for fetch : https://github.com/razorpay/razorpay-node/blob/master/documents/items.md#fetch-particular-item
Link : https://razorpay.com/docs/api/items/#fetch-an-item

for all : https://github.com/razorpay/razorpay-node/blob/master/documents/items.md#fetch-all-items
Link : https://razorpay.com/docs/api/items/#fetch-multiple-items

for edit : https://github.com/razorpay/razorpay-node/blob/master/documents/items.md#update-item
Link : https://razorpay.com/docs/api/items/#update-an-item

for delete : https://github.com/razorpay/razorpay-node/blob/master/documents/items.md#delete-item
Link : https://razorpay.com/docs/api/items/#delete-an-item

### CODE REVIEW

File: test/fixtures/fake_item.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    "type": "invoice",
+    "unit": null,
+    "tax_inclusive": false,
+    "hsn_code": null,
+    "sac_code": null,
+    "tax_rate": null,
+    "tax_id": null,
+    "tax_group_id": null,
+    "created_at": 1647584637
+  }
```
Comment (arjunmenon) [inline test/fixtures/fake_item.json:18]: Representation of item entity not according to doc - https://razorpay.com/docs/api/items/#create-an-item

File: lib/razorpay/item.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    def self.fetch(id)
+      request.fetch id
+    end
+
+    def self.edit(id, options = {})
+      if(!options.is_a?(String) && options.key?(:active))
+        options[:active] = (options[:active] ? 1 : 0)
+      end 
+      request.patch id, options
```
Comment (arjunmenon) [inline lib/razorpay/item.rb:23]: Updating url is `item/{id}`

File: lib/razorpay/item.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    end
+
+    def self.fetch(id)
+      request.fetch id
+    end
+
+    def self.edit(id, options = {})
+      if(!options.is_a?(String) && options.key?(:active))
+        options[:active] = (options[:active] ? 1 : 0)
+      end
```
Comment (arjunmenon) [inline lib/razorpay/item.rb:22]: Why needed? based on docs, they are expected to send only true or false. Dont think we should transpose

File: lib/razorpay/item.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    def self.edit(id, options = {})
+      if(!options.is_a?(String) && options.key?(:active))
+        options[:active] = (options[:active] ? 1 : 0)
+      end 
+      request.patch id, options
+    end
+
+    def self.all(options = {})
+      request.all options
+    end
```
Comment (arjunmenon) [inline lib/razorpay/item.rb:28]: Based on docs, https://razorpay.com/docs/api/items/#fetch-multiple-items, what options are we expecting?

File: lib/razorpay/item.rb (hunk 3 above)
Comment (ankitdas13) [reply lib/razorpay/item.rb:22]: in some sdk like php and python its working fine but in node and ruby when user put `true/false` it shows message `he active field must be true or false.` .  i used same method for [node](https://github.com/razorpay/razorpay-node/blob/master/lib/resources/items.js#L80)

File: test/razorpay/test_item.rb [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    def test_edit_items
+      
+       param_attr = {
+        "name": "Book / English August",
+        "description": "An indian story, Booker prize winner.",
+        "amount": 20000,
+        "currency": "INR"
+       }
+
+      stub_patch(%r{items/#{@item_id}$}, 'fake_item', param_attr.to_json)
```
[the remark this answers is not in the document]
Comment (ankitdas13) [reply test/razorpay/test_item.rb:42]: url is correct https://razorpay.com/docs/api/items/#request-parameters-1

File: lib/razorpay/item.rb (hunk 4 above)
Comment (ankitdas13) [reply lib/razorpay/item.rb:28]: its working with parameter like `from, to, count, active` like other `all` api

File: lib/razorpay/item.rb (hunk 2 above)
Comment (ankitdas13) [reply lib/razorpay/item.rb:23]: its already there https://github.com/razorpay/razorpay-ruby/blob/item_update/lib/razorpay/item.rb#L8 while requesting it append

File: test/fixtures/fake_item.json (hunk 1 above)
Comment (ankitdas13) [reply test/fixtures/fake_item.json:18]: Done

File: lib/razorpay/item.rb (hunk 2 above)
Comment (arjunmenon) [reply lib/razorpay/item.rb:23]: According to doc, its `item/{id}`. What you have implemented is `items/{id}`

File: test/razorpay/test_item.rb (hunk 5 above)
[the remark this answers is not in the document]
Comment (arjunmenon) [reply test/razorpay/test_item.rb:42]: Pls rechk docs or confirm from API owner. In docs it is written as `item/{id}`

File: lib/razorpay/item.rb (hunk 4 above)
Comment (arjunmenon) [reply lib/razorpay/item.rb:28]: Pls confirm from API owner

File: lib/razorpay/item.rb (hunk 2 above)
Comment (ankitdas13) [reply lib/razorpay/item.rb:23]: Screenshot 2022-04-06 at 5 39 38 PM
i checked that . the doc need to be correction. if you check in code snippet `curl` its `items/{id}`

File: test/razorpay/test_item.rb (hunk 5 above)
[the remark this answers is not in the document]
Comment (ankitdas13) [reply test/razorpay/test_item.rb:42]: the doc need to be correction. if you check in code snippet `curl` its `items/{id}`

File: lib/razorpay/item.rb (hunk 2 above)
Comment (ankitdas13) [reply lib/razorpay/item.rb:23]: need to confirm link

**Review by arjunmenon — APPROVED** [review]
\<internal\> Approving but pls update Rzp public doc. [Thread](https://razorpay.slack.com/archives/C029ZL11PNK/p1649330587325239?thread_ts=1647927334.987909&cid=C029ZL11PNK).

**Pull request merged**